### PR TITLE
Replace lazy_static with once_cell

### PR DESCRIPTION
**Pull request by dralley**
The once_cell API is currently being incorporated into the stdlib.  It's macro-less and more powerful.

https://github.com/rust-lang/rust/issues/74465

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -9,15 +9,15 @@ keywords      = ["rpm", "linux", "redhat", "fedora", "centos"]
 readme        = "README.md"
 homepage      = "https://github.com/rpm-software-management/librpm.rs/"
 documentation = "https://rustrpm.org/librpm/"
-edition       = "2018"
+edition       = "2021"
```
Comment (tarcieri) [inline Cargo.toml:12]: I'm okay with an edition bump, but IMO it should be done in a separate PR that notes the corresponding MSRV change

**Comment by dralley** [comment]
addressed
